tcms/testplans/tests/test_edit_plan.py

```diff
@@ -0,0 +1,65 @@
+from http import HTTPStatus
+
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from tcms.tests import user_should_have_perm, BasePlanCase
+
+
+class EditPlanViewTest(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.testplan_edit_url = reverse('plan-edit', args=[cls.plan.pk])
+
+        cls.testplan_edit_data = {
+            'author': cls.tester.pk,
+            'product': cls.plan.product.pk,
+            'product_version': cls.plan.product_version.pk,
+            'type': cls.plan.type.pk,
+            'name': cls.plan.name,
+
+            'email_settings-0-auto_to_plan_author': 'on',
+            'email_settings-0-auto_to_case_owner': 'on',
+            'email_settings-0-auto_to_case_default_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_plan_update': 'on',
+
+            'email_settings-0-plan': '',
+            'email_settings-0-id': cls.plan.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        }
+
+        user_should_have_perm(cls.tester, perm='testplans.change_testplan')
+
+    def test_show_testplan_edit_page(self):
+        response = self.client.get(self.testplan_edit_url)
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertContains(response, self.plan.name)
+        self.assertContains(response, _('Edit TestPlan'))
+        self.assertContains(response,
+                            'name="email_settings-TOTAL_FORMS" value="1"')
+        self.assertContains(response,
+                            'name="email_settings-INITIAL_FORMS" value="1"')
+        self.assertContains(response,
+                            'name="email_settings-MIN_NUM_FORMS" value="0"')
+        self.assertContains(response,
+                            'name="email_settings-MAX_NUM_FORMS" value="1"')
+        self.assertContains(response,
+                            'name="email_settings-TOTAL_FORMS" value="1"')
```

Comment:
These asserts should check the values of the on/off buttons, they are more important than these hidden fields, see previous comment.